addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -287,8 +327,18 @@ private ApplicationMatch checkCookieMatches(
             for (Map.Entry<String, AppPattern> entry : sp.entrySet()) {
                 for (HtmlParameter cookie : msg.getCookieParams()) {
                     if (entry.getKey().equals(cookie.getName())) {
-                        AppPattern p = entry.getValue();
-                        appMatch = addIfMatches(appMatch, currentApp, p, cookie.getValue());
+                        if (entry.getValue().getJavaPattern().toString().isEmpty()
+                                && entry.getValue().getRe2jPattern().toString().isEmpty()) {
```

Comment:
e.g. `!isCookieValueCheck(…)` and the extra code comments are no longer needed. This should have tests.

Reply:
Test was not added?

Reply:
`shouldNotMatchOnlyCookieNameInRequestWhenValueIsAlsoExpected` & `shouldNotMatchOnlyCookieNameWhenValueIsAlsoExpected` were already there. I've now added: `shouldMatchCookieNameAndValueInRequestWhenBothExpected`

Reply:
I've also discovered that headers had the same issue and have changed code and added tests.

Reply:
That's already covered by `shouldMatchCookieNameAndValueInRequestWithTestEntry`. Both `shouldMatchCookieNameInRequestWithPhp` and `shouldMatchCookieNameWithPhp` cover the case of cookie without value.